client/components/carousel.jsx

```diff
@@ -0,0 +1,169 @@
+            <div className="caro-content-holder">
+              <div className="col-two-fifth caro-pic-col">
+                <h3 className="mg-t-75">{favorite.favoriteDetails.strAlbum}</h3>
+                <div className="caro-pic-holder">
+                  <img src={favorite.favoriteDetails.strAlbumThumb} />
+                </div>
+              </div>
+              <div className="col-two-fifth caro-info-col">
+                <div className="caro-detail-bt-row"><button className="caro-detail-button">Detail</button></div>
+                <h3 className="caro-text">Artist: <span>{favorite.favoriteDetails.strArtist}</span></h3>
+                <h3 className="caro-text">First Release: <span>{favorite.favoriteDetails.intYearReleased || 'N/A'}</span></h3>
+                <h3 className="caro-text">Genre: <span>{favorite.favoriteDetails.strGenre || 'N/A'}</span></h3>
+                <h3 className="caro-text">Style: <span>{favorite.favoriteDetails.strStyle || 'N/A'}</span></h3>
+                <h3 className="caro-text">Album Score: <span>{favorite.favoriteDetails.intScore || 'N/A'}</span></h3>
+              </div>
+            </div>
+          </div>
+        );
+      }
+      if (favorite.favoriteType === 'artist') {
+        return (
+          <div key={favorite.favoriteId} className={index === this.state.index ? '' : 'hidden'}>
+            <div className="caro-content-holder">
+              <div className="col-two-fifth caro-pic-col">
+                <h3 className="mg-t-75">{favorite.favoriteDetails.strArtist}</h3>
+                <div className="caro-pic-holder">
+                  <img src={favorite.favoriteDetails.strArtistThumb} />
+                </div>
+              </div>
+              <div className="col-two-fifth caro-info-col">
+                <div className="caro-detail-bt-row"><button className="caro-detail-button">Detail</button></div>
+                <h3 className="caro-text ">Alternate Name: <span>{favorite.favoriteDetails.strArtistAlternate || 'N/A'}</span></h3>
+                <h3 className="caro-text">Born: <span>{favorite.favoriteDetails.intBornYear || 'N/A'}</span></h3>
+                <h3 className="caro-text">Genre: <span>{favorite.favoriteDetails.strGenre || 'N/A'}</span></h3>
+                <h3 className="caro-text">Style: <span>{favorite.favoriteDetails.strStyle || 'N/A'}</span></h3>
+                <h3 className="caro-text artist-web">Artist Website: <span>{favorite.favoriteDetails.strWebsite || 'N/A'}</span></h3>
+              </div>
+            </div>
+          </div>
+
+        );
+      }
+      return null;
+    });
+
+    const listButtons = favorites.map((favorite, index) =>
+      <button onClick={this.dotClick} className={index === this.state.index ? 'dot dot-select' : 'dot'} name={index} key={favorite.favoriteId}></button>
+    );
+    const carouselDisplay = favorites ? favoriteLists : defaultDisplay;
```

Comment:
Creating all of those React elements and then ignoring a lot of them is very inefficient, try to reorganize the code into separate components and pass them props. This is much more efficient:

```jsx
const carouselDisplay = favorites ? <Favorites /> : <DefaultDisplay />;
```